### PR TITLE
feat: add support to push logs to opensearch

### PR DESCRIPTION
**Pull request by prasunna09**

### CODE REVIEW

File: lib/aws/log_stack.ts [hunk 1]
```diff
@@ -0,0 +1,402 @@
...
+        const kAnalyticsNS = cluster.addManifest("kube-analytics-ns", {
+            "apiVersion": "v1",
+            "kind": "Namespace",
+            "metadata": {
+                "name": "kube-analytics"
+            }
+        });
+
+        kAnalyticsNS.node.addDependency(this.domain);
+        let open_search_master_user_name = scope.node.tryGetContext('open_search_master_user_name');
```
Comment (nitesh-balla) [inline lib/aws/log_stack.ts:185]: What happens if this isn't passed in the cdk deploy command, Can we handle it please?

File: lib/aws/log_stack.ts [hunk 2]
```diff
@@ -0,0 +1,402 @@
...
+        let open_search_master_password = scope.node.tryGetContext('open_search_master_password');
```
Comment (nitesh-balla) [inline lib/aws/log_stack.ts:186]: Same here

File: install.sh [hunk 3]
```diff
@@ -363,6 +363,66 @@ while true; do
     fi
 done
 
+validate_master_password() {
```
Comment (nitesh-balla) [inline install.sh:366]: Let us change the function name to `validate_opensearch_password`

File: lib/aws/eks.ts [hunk 4]
```diff
@@ -55,7 +56,11 @@ export class EksStack {
       vpc: vpc,
       clusterName: "hs-eks-cluster",
     });
-
+    let open_search_service = scope.node.tryGetContext('open_search_service');
```
Comment (nitesh-balla) [inline lib/aws/eks.ts:59]: Same here

File: install.sh (hunk 3 above)
Comment (nitesh-balla) [reply install.sh:366]: Can we try to reuse the existing function if possible?

**Review by nitesh-balla — CHANGES_REQUESTED** [review]

File: install.sh (hunk 3 above)
Comment (prasunna09) [reply install.sh:366]: both have different conditions

**Review by nitesh-balla — APPROVED** [review]

**Pull request merged**